File: perl/leapyear.pl
#!/usr/bin/perl -w
##############################################################################
#    Name: leap
#
# Summary: Determine if year passed to this pgm is a leap year.
#
# Created: Mon 26 Feb 2001 12:14:07 (Bob Heckel)
##############################################################################

if ( $#ARGV ) {
  print "Usage: $0 year\n";
  exit(-1);
}

###$year = 1800;    # no
###$year = 2000;    # yes
$year = $ARGV[0];

if ( (!($year % 4) && ($year % 100)) || !($year % 400) ) {
  print "$year is a leap year.\n";
} else {
  print "$year is not a leap year.\n";
}

File: misc/c/leapyear.c
/*****************************************************************************
 *     Name: leapyear.c
 *
 *  Summary: Determine if year is a leap year.
 *
 *  Adapted: Mon 08 Jul 2002 16:21:43 (Bob Heckel -- freshsources.com)
 *****************************************************************************
*/
#include <stdio.h>

inline int isleap(int y) { return y%4 == 0 && y%100 != 0 || y%400 == 0;};

int main(int argc, char *argv[]) {
  if ( isleap(2000) ) {
    puts("year is a leap year");
  } else {
    puts("year is a not leap year");
  }

  return 0;
}
